src/libaktualizr/uptane/uptanerepository.cc

```diff
@@ -67,52 +75,28 @@ bool RepositoryCommon::updateRoot(INvStorage& storage, const IMetadataFetcher& f
     }
   }
 
-  // Update to latest Root metadata.
-  {
-    // According to the current design root.json without a number is guaranteed to be the latest version.
-    // Therefore we fetch the latest (root.json), and
-    // if it matches what we already have stored, we're good.
-    // If not, then we have to go fetch the missing ones by name/number until we catch up.
-
+  // 5.4.4.3.2. Update to the latest Root metadata file.
+  for (int version = rootVersion() + 1;; ++version) {
```

Comment:
Should this loop be upper-bounded? It seems a bit dangerous to never stop here if the server keeps supplying valid metadata.

Reply:
That's a really good point. Yes, it probably should be, but what's reasonable? 100? We should probably bring this up directly with the Uptane group. Or do you have an opinion?

Reply:
It sounds like another limit implementations should enforce, like the size of downloaded metadata artifacts.

I am thinking if we should have a limit for version jumps during an update or just a global upper bound (which we need anyway since we're using json integers and an `int` in this code).

The "jump" limit might complicate diagnostic of accidental trigger, since it would only affect a fraction of the devices.

Reply:
> It sounds like another limit implementations should enforce, like the size of downloaded metadata artifacts.

Could be, yes.

> I am thinking if we should have a limit for version jumps during an update or just a global upper bound (which we need anyway since we're using json integers and an `int` in this code).

Global might make more sense, because if you provision a new device, it'll have to catch up the whole way to the latest anyway. But I'm also a bit concerned that it might be unexpected to have such a limit.

I think we should either leave it is or make a quick decision now, and either way make a ticket to follow up on it and in the meantime move forward with getting this merged. Do you have a preference? If we want to just make a quick decision, I'd say 100 should be fine for now.

Reply:
Ok I see, then 100 or 1000 would be fine for now.

Reply:
Done (with 1000 for now), and made https://saeljira.it.here.com/browse/OTA-4273 to re-evaluate again.